### PR TITLE
Action for checking dead links

### LINKED ISSUE #1977: Automate Dead Link Detection in Docs
- https://github.com/marketplace/actions/broken-link-check

### PR DESCRIPTION
**Pull request by ozzieD**
- Fixes #1977 
- We want an automated way to check for deadlinks in our documentation. This MR introduces an action that checks [our latest documentation](https://featuretools.alteryx.com/en/latest/) for deadlinks recursively.

### CODE REVIEW

File: docs/source/release_notes.rst [hunk 1]
```diff
@@ -20,6 +20,7 @@ Future Release
         * Rename yml to yaml for GitHub Actions workflows (:pr:`2073`, :pr:`2077`)
         * Update Dask test fixtures to prevent flaky behavior (:pr:`2079`)
         * Update Makefile with better pkg command (:pr:`2081`)
+        * Add scheduled workflow that checks for broken links in documentation (:pr:`2083`)
 
     Thanks to the following people for contributing to this release:
     :user:`gsheni`, :user:`rwedge`, :user:`thehomebrewnerd`
```
[the remark this answers is not in the document]
Comment (gsheni) [reply docs/source/release_notes.rst:26]: Add yourself to list of contributors

File: .github/workflows/broken_link_check.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    steps:
+      - name: Check for broken links
+        id: link-report
+        uses: celinekurpershoek/link-checker@v1.0.1
+        with:
+          # Required:
+          url: "https://featuretools.alteryx.com/en/latest/"
+          # optional:
+          honorRobotExclusions: true
+          ignorePatterns: "*github*,*google*,*twitter*,*slack*,*stackoverflow*,*anaconda*"
```
Comment (gsheni) [inline .github/workflows/broken_link_check.yml:19]: Why are we ignoring this?

Comment (ozzieD) [reply .github/workflows/broken_link_check.yml:19]: I can put some of these back in but a couple of these produced false positives like the pull requests in release notes.

File: .github/workflows/broken_link_check.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+name: Broken link check
+on:
+  schedule:
```
Comment (gsheni) [inline .github/workflows/broken_link_check.yml:3]: - Add this so we can kickoff the job manually from here (https://github.com/alteryx/featuretools/actions)
```suggestion
  workflow_dispatch:
  schedule:
```

**Pull request closed without merging**